src/lib/writers/OutputFileWriter.ts

```diff
@@ -0,0 +1,44 @@
+import * as fs from 'node:fs';
+import {OutputFormat, RunResults} from '@salesforce/code-analyzer-core';
+import {BundleName, getMessage} from '../messages';
+
+export interface OutputFileWriter {
+	writeToFiles(results: RunResults): void;
+	getOutputFiles(): string[];
+}
+
+export class OutputFileWriterImpl implements OutputFileWriter {
+	private outputFilesToFileFormat: Map<string, OutputFormat> = new Map();
+
+	public constructor(files: string[]) {
+		for (const file of files) {
+			if (file.endsWith(`.csv`)) {
```

Comment:
`path.ext(blah.sarif.json)` just returns `.json`, which is why I didn't use it here. But it could definitely be used for the depth-1 extensions. I'll get on that.